src/views/AddRecord.jsx

```diff
@@ -8,17 +8,84 @@ import { MaterialIcons } from "@expo/vector-icons";
 import MapView, { Marker } from "react-native-maps";
 import { AddRecordStyle } from "../themes/AddRecordStyle";
 import { mapDark } from "../themes/MapStyle";
+import * as ImagePicker from 'expo-image-picker';
+import { auth, storage, db} from '../api/firebase'
+import { doc, collection, query, getDocs, setDoc} from 'firebase/firestore'
+import { ref, uploadBytes } from 'firebase/storage'
+import 'react-native-get-random-values';
+import { v4 as uuidv4 } from 'uuid';
+import ButtonBlue from "../components/ButtonBlue";
+
+
 
 export class AddRecord extends Component {
-  state = {
-    markerLocation: null,
+  constructor(props) {
+    super(props);
+    this.state = {
+      image: null,
+      isLoading: false,
+      uploading: false,
+      descripcion: '',
+      markerLocation: {
+                latitude: 19.244,
+                longitude: -103.725
+      }
+    };
+  }
+
+  pickImage = async () => {
+    let result = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.All,
+      allowsEditing: true,
+      aspect: [4, 3],
+      quality: 1,
+    });
+    const source = { uri: result.assets[0].uri };
+    this.setState({ image: source });
   };
+  
+  uploadImage = async () => {
+    this.setState({ uploading: true });
+    const response = await fetch(this.state.image.uri);
+    const blob = await response.blob();
+    const id = uuidv4();
+    const refImage = ref(storage, `${auth.currentUser.uid}/${id}`);
+    try {
+    await  uploadBytes(refImage, blob)
+    this.setState({ uploading: false, image: null });
+    
+    const reporte = {
+      descripcion: this.state.descripcion,
+      idImg: id,
+      locacion: this.state.markerLocation,
+      fecha: new Date(),
+    };
+    
+      const newReporte = doc(collection(db, `users/${auth.currentUser.uid}/reports`));
+      const q = query(collection(db, 'users'));
```

Comment:
Could you please give a better name for this const?

Reply:
done